src/libraries/System.Drawing.Common/src/System/Drawing/ImageInfo.cs

```diff
@@ -127,15 +130,15 @@ public bool FrameDirty
             }
 
             /// <summary>
-            /// The total animation time of the image, in milliseconds.
+            /// The total animation time of the image in milliseconds, or <value>0</value> if not animated.
             /// </summary>
-            private long TotalAnimationTime => Animated ? _frameEndTimes![_frameCount - 1] : 0;
+            private long TotalAnimationTime => Animated ? _totalAnimationTime : 0;
 
             /// <summary>
             /// Whether animation should progress, respecting the image's animation support
             /// and if there are animation frames or loops remaining.
             /// </summary>
-            private bool ShouldAnimate => Animated ? (_loopCount == 0 || _loop <= _loopCount) : false;
+            private bool ShouldAnimate => TotalAnimationTime > 0 ? (_loopCount == 0 || _loop <= _loopCount) : false;
```

Comment:
This changes the behavior at line 160, preventing animation from occurring if the total animation time is <= 0.